messaging/broker.go

```diff
@@ -92,25 +126,67 @@ func (b *Broker) Open(path string, u *url.URL) error {
 	}
 	b.path = path
 
-	// Require a non-blank connection address.
-	if u == nil {
-		return ErrConnectionAddressRequired
+	// Ensure root directory exists.
+	if err := os.MkdirAll(path, 0700); err != nil {
+		b.close()
+		return fmt.Errorf("mkdir: %s", err)
+	}
+
+	// Open meta file.
+	meta, err := bolt.Open(b.metaPath(), 0600, &bolt.Options{Timeout: 1 * time.Second})
+	if err != nil {
+		b.close()
```

Comment:
I've started using the following pattern in my code, so take care of this kind of stuff, since `b.close()` is called from multiple places and it can be tedious and error-prone to add it everywhere.

- Name the error i.e. `err error`
- then at the start of the function:
```
defer func() {
    if err != nil && b != nil {
        b.close()
    }
}()
```
Thoughts?